classes/hook/filter_sections.php

```diff
@@ -0,0 +1,193 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Hook class for filtering a list of sections.
+ *
+ * @copyright  2024 ISB Bayern
+ * @author     Philipp Memmel
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace block_massaction\hook;
+
+use coding_exception;
+
+defined('MOODLE_INTERNAL') || die();
+
+#[\core\attribute\label('Hook dispatched whenever block_massaction is using a list of sections.')]
+#[\core\attribute\tags('block_massaction')]
+/**
+ * Hook class for filtering a list of sections.
+ *
+ * @copyright  2024 ISB Bayern
+ * @author     Philipp Memmel
+ * @package    block_massaction
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class filter_sections {
+
+    /** @var array Array of section numbers which originally are available by block_massaction. */
+    private readonly array $originalsectionnums;
+
+    /** @var bool Determines if the user will be able to keep the original section of a course module when performing some operations. */
+    private bool $keeporiginalsectionallowed;
```

Comment:
As a typed property, does this need to be initialized somewhere?  I would think the is_keeporiginalsectionallowed() method would fail on actions in SAMECOURSE.  Should it be set to false by default?

Reply:
Yeah, this was because of the combined designed of both hooks in one. I changed that design but also included your feedback.